### PR TITLE
#161290954 setup redux

### PR DESCRIPTION
**Pull request by augustineezinwa**
#### What does this PR do?
- This PR setups the redux aspect of the application.
- Components were also connected to store and tested
#### Description of Task to be completed?
- Install redux
- Install react-redux
- Configure redux-dev-tools
- Configure store
- Connect components to store
- Render dummy component
- Write tests
#### How should this be manually tested?
- `npm test`
#### Any background context you want to provide?
- None
#### What are the relevant pivotal tracker stories?
-[ #161290954](https://www.pivotaltracker.com/story/show/161290954)
#### Screenshots (if appropriate)
screen shot 2018-12-01 at 7 17 38 pm
screen shot 2018-12-01 at 7 19 51 pm
screen shot 2018-12-01 at 7 20 19 pm

#### Questions:
- N/A

### CODE REVIEW

**Review by darthrighteous — COMMENTED** [review]
There are a number of hound violations, fix those. Also, this is raised against master, fix that to develop.

**Comment by sulenchy** [comment]
@Darthrighteous feedback duly noted.

**Comment by sulenchy** [comment]
@Darthrighteous feedback implemented. No violation now

**Review by tersoo-atsen — APPROVED** [review]

File: src/App.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+import { Provider } from 'react-redux';
+import Button from './containers/Button';
+import store from './store/store';
+
+const App = () => (
+  <Provider store ={store}>
+  <div className='jumbotron'>
```
Comment (mbilesanmi) [inline src/App.js]: Your indentation seems to be way off in this file.

File: src/reducers/buttonReducer.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import initialState from '../store/initialState';
+import * as types from '../actions/actionTypes';
+
+const { button } = initialState;
+const buttonReducer = (state = button, action) => {
+  const { type } = action;
+  const { ON_BUTTON, OFF_BUTTON } = types;
```
Comment (mbilesanmi) [inline src/reducers/buttonReducer.js]: Move this to the import and import only the required types.

**Review by mbilesanmi — CHANGES_REQUESTED** [review]
Please fix the 2 issues and it looks good for me. Then rebase @augustineezinwa

**Review by mbilesanmi — APPROVED** [review]

**Review by KvNGCzA — APPROVED** [review]

**Comment by mbilesanmi** [comment]
1. Why do we still have conflicts?
2. How do we have up to 40 files changed?
3. Why are we having `*.snap` files showing up?
4. Are we squashing commits, or how is it that there is only 1 commit despite the magnitude of file changes we have?

@augustineezinwa @sulenchy

**Comment by sulenchy** [comment]
> 1. Why do we still have conflicts?
> 2. How do we have up to 40 files changed?
> 3. Why are we having `*.snap` files showing up?
> 4. Are we squashing commits, or how is it that there is only 1 commit despite the magnitude of file changes we have?
> 
> @augustineezinwa @sulenchy
1. The conflict has been resolved now
2. We are not having the `*.snap` files again
3. we used `git commit --amend`

**Comment by augustineezinwa** [comment]
> 1. Why do we still have conflicts?
> 2. How do we have up to 40 files changed?
> 3. Why are we having `*.snap` files showing up?
> 4. Are we squashing commits, or how is it that there is only 1 commit despite the magnitude of file changes we have?
> 
> @augustineezinwa @sulenchy

Everything is fine now

**Comment by augustineezinwa** [comment]
> 1. Why do we still have conflicts?
> 2. How do we have up to 40 files changed?
> 3. Why are we having `*.snap` files showing up?
> 4. Are we squashing commits, or how is it that there is only 1 commit despite the magnitude of file changes we have?
> 


@mbilesanmi Everything is fine now

**Pull request merged**